packages/core/src/lib/core/queries/resolvers.ts

```diff
@@ -106,6 +106,18 @@ export async function findOne(
   return runWithPrisma(context, list, model => model.findFirst({ where: filter }));
 }
 
+export async function findSingleton(list: InitialisedSingleton, context: KeystoneContext) {
+  // Check operation permission to pass into single operation
+  const operationAccess = await getOperationAccess(list, context, 'query');
+  if (!operationAccess) {
+    return null;
+  }
+
+  // TODO add item access control
```

Comment:
As discussed during the review, we want to remove `findSingleton` and `updateSingleton`,  but notably can we add some tests for access control with Singletons